src/clp_ffi_js/ir/LogEventWithFilterData.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+#define CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+
+#include <utility>
+
+#include <clp/ir/types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+/**
+ * A class which accepts a log event type as a template parameter and provides additional members
+ * for the log level and timestamp. The additional members facilitate log level filtering.
+ * @tparam LogEvent The type of the log event.
+ */
+template <typename LogEvent>
+class LogEventWithFilterData {
+public:
+    // Constructor
+    LogEventWithFilterData(
+            LogEvent log_event,
+            LogLevel log_level,
+            clp::ir::epoch_time_ms_t timestamp
+    )
+            : m_log_event{std::move(log_event)},
+              m_log_level{log_level},
+              m_timestamp{timestamp} {}
+
+    // Disable copy constructor and assignment operator
```

Comment:
Why?

Reply:
> The key value pair log event has it.

Yeah, I disagree with that decision, lol. I will follow up on that separately.

> So felt it made sense to have it here.

Fair enough. I will come back to this once we resolve the issue with KeyValuePairLogEvent.

> It also warns people if they are trying to copy it when it can be moved.

I think that argument is a bit of a slippery slope unless you have a type that is truly too massive to be copied. Yes, we want users to write high-performance code, but trying to force them to do that by disabling copying just hurts users who really want a copy of an object. That user may then want to add a clone member to your class at which point we're just adding syntax sugar for the sake of preventing code that *might* perform poorly if the compiler doesn't already optimize it.